packages/loopring_v3/contracts/core/iface/IExchangeV3.sol

```diff
@@ -85,7 +85,8 @@ abstract contract IExchangeV3 is Claimable
 
     event TransactionApproved(
         address owner,
-        bytes32 transactionHash
+        bytes32 transactionHash,
+        ExchangeData.ApprovalType approvalType
```

Comment:
Maybe we can emit msg.sender here and use it to check where the approval came from.